hera_cal/io.py

```diff
@@ -0,0 +1,286 @@
+        antpos, ants = uvd.get_ENU_antpos(center=True, pick_data_ants=pick_data_ants)
+        antpos = odict(zip(ants, antpos))
+        pols = np.array([polnum2str(polnum) for polnum in uvd.polarization_array])
+        return data, flags, antpos, ants, freqs, times, lsts, pols
+    else:
+        return data, flags
+
+
+def write_vis(outfilename, data, flags, format='miriad', history='', clobber=False, **kwargs):
+    '''TODO: migrate in hera_cal.utils.data_to_miriad and generalize to also write uvfits.'''
+    raise NotImplementedError('This function has not been implemented yet.')
+
+
+def update_vis(infilename, outfilename, format_in='miriad', format_out='miriad', 
+               data=None, flags=None, add_to_history='', clobber=False, **kwargs):
+    '''Loads an existing file with pyuvdata, modifies some subset of of its parameters, and 
+    then writes a new file to disk. Cannot modify the shape of data arrays. More than one 
+    spectral window is not supported.
+
+    Arguments:
+        infilename: filename of the base visibility file to be updated, or UVData object
+        outfilename: filename of the new visibility file
+        format_in: either 'miriad' or 'uvfits' (ignored if infile is a UVData object)
+        format_out: either 'miriad' or 'uvfits'
+        data: dictionary or DataContainer of complex visibility data to update. Keys
+            like (0,1,'xx') and shape=(Ntimes,Nfreqs). Default (None) does not update. 
+        flags: dictionary or DataContainer of data flags to update. 
+            Default (None) does not update.
+        add_to_history: prepends a string to the history of the output file
+        clobber: if True, overwrites existing file at outfilename
+        kwargs: dictionary mapping updated attributs to their new values. 
+            See pyuvdata.UVData documentation for more info.
+    '''
+
+    # Load infile
+    if type(infilename) == UVData:
+        uvd = deepcopy(infilename)
+    else:    
+        uvd = UVData()
+        if format_in is 'miriad':
+            uvd.read_miriad(infilename)
+        elif format_in is 'uvfits':
+            #TODO: implement this
+            raise NotImplementedError('This function has not been implemented yet.')
+        else:
+            raise TypeError("Input format must be either 'miriad' or 'uvfits'.")
+
+    # set data and/or flags
+    if data is not None or flags is not None:
+        bls = map(uvd.baseline_to_antnums, np.unique(uvd.baseline_array))
+        for (i,j) in bls:
+            this_bl = (uvd.baseline_array == uvd.antnums_to_baseline(i,j))
+            for ip, pol in enumerate(uvd.polarization_array):
+                if data is not None:
+                    uvd.data_array[this_bl,0,:,ip] = data[(i,j,polnum2str(pol))]
+                if flags is not None:
+                    uvd.flag_array[this_bl,0,:,ip] = flags[(i,j,polnum2str(pol))]
+
+    # set additional attributes
+    uvd.history = add_to_history + uvd.history
```

Comment:
I don't think doing that really gives us a lot of information, since it doesn't tell us how it was altered. Let's just be thorough in making sure that everything that uses this function in the pipeline actually provides a meaningful string here.

Reply:
perhaps switch order such that `uvd.history = hvd.history + add_to_history`. that makes more chronological sense (I know that's how I do it in `data_to_miriad` but been meaning to switch it)

Reply:
Good call. I'll do that.